### PR TITLE
Allow offline/local servers to keep skins for users with accounts

### PR DESCRIPTION
**Pull request by ghost**
Say I want to play with friends, but a couple people don't have a mineshafter squared account. It's a "private" server, so authentication isn't really an issue but I want the players with mineshafter accounts to still have their skins (no skins in offline-mode)

This modifies the game controller (join_server and check_server). Not sure I did check_server right; I'm guessing it's to confirm the user-joining was logged properly in the database? If so, that validity check is not needed for unauthenticated servers (the server admin just trusts only his friends are coming on and the proxy is just so the users with accounts get their skins)

Basically, the GUI would let the server admin list names he/she wants to whitelist and it would append the noverify param to the proxy URL.

### CODE REVIEW

**Comment by alexander12366** [comment]
i allow
